PWGHF/HFC/TableProducer/HFCorrelatorD0Hadrons.cxx

```diff
@@ -0,0 +1,491 @@
+  //////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+  void processMcGen(aod::McCollision const& mccollision, soa::Join<aod::McParticles, aod::HfCandProng2MCGen> const& particlesMC)
+  {
+
+    registry.fill(HIST("hMCEvtCount"), 0);
+    // MC gen level
+    for (auto& particle1 : particlesMC) {
+      // check if the particle is D0 or D0bar (for general plot filling and selection, so both cases are fine) - NOTE: decay channel is not probed!
+      if (std::abs(particle1.pdgCode()) != pdg::Code::kD0) {
+        continue;
+      }
+      double yD = RecoDecay::y(array{particle1.px(), particle1.py(), particle1.pz()}, RecoDecay::getMassPDG(particle1.pdgCode()));
+      if (cutYCandMax >= 0. && std::abs(yD) > cutYCandMax) {
+        continue;
+      }
+      if (cutPtCandMin >= 0. && particle1.pt() < cutPtCandMin) {
+        continue;
+      }
+      registry.fill(HIST("hPtCandMCGen"), particle1.pt());
+      registry.fill(HIST("hEtaMCGen"), particle1.eta());
+      registry.fill(HIST("hPhiMCGen"), particle1.phi());
+      registry.fill(HIST("hYMCGen"), yD);
+
+      // D-h correlation dedicated section
+
+      if (std::abs(particle1.pdgCode()) != pdg::Code::kD0) // just checking the particle PDG, not the decay channel (differently from Reco: you have a BR factor btw such levels!)
+        continue;
+      registry.fill(HIST("hCountD0triggersMCGen"), 0, particle1.pt()); // to count trigger D0 (for normalisation)
+
+      for (auto& particle2 : particlesMC) {
+        registry.fill(HIST("hTrackCounterMCGen"), 1); // total no. of tracks
+        if (std::abs(particle2.eta()) > cutTrackEtaMax)
+          continue;
+
+        if (particle2.pt() < cutPtTrackMin)
+          continue;
+
+        if ((std::abs(particle2.pdgCode()) != kElectron) && (std::abs(particle2.pdgCode()) != kMuonMinus) && (std::abs(particle2.pdgCode()) != kPiPlus) && (std::abs(particle2.pdgCode()) != kKPlus) && (std::abs(particle2.pdgCode()) != kProton))
+          continue;
+
+        // ==============================soft pion removal================================
+        registry.fill(HIST("hTrackCounterMCGen"), 2); // fill before soft pi removal
+        // method used: indexMother = -1 by default if the mother doesn't match with given PID of the mother. We find mother of pion if it is D* and mother of D0 if it is D*. If they are both positive and they both match each other, then it is detected as a soft pion
+
+        auto indexMotherPi = RecoDecay::getMother(particlesMC, particle2, 413, true, nullptr, 1); // last arguement 1 is written to consider immediate decay mother only
+        auto indexMotherD0 = RecoDecay::getMother(particlesMC, particle1, 413, true, nullptr, 1);
```

Comment:
Still hard-coded values.

Reply:
Hi Vit, Please suggest me what should I write for Dstar instead of 413, I couldn't find anything suitable.

Reply:
Hi @samrangy , you have to define it in `Core/HFSelectorCuts.h` then.

Reply:
There is a conflict in merging HFSelectorCuts.h, seems I don't have write access.
Also, I'm getting error in MegaLinter, is it due to the same file?

Reply:
What do you mean you don't have write access? You have to update your local repository, resolve the conflicts and push to your repository on GitHub.
The MegaLinter error is unrelated and will go away after you update your branch.